microsoft/testsuites/dpdk/dpdktestpmd.py

```diff
@@ -0,0 +1,164 @@
+        else:
+            return False
+
+    def _install_dependencies(self) -> bool:
+        node = self.node
+        cwd = node.working_path
+        if isinstance(node.os, Ubuntu):
+            node.os.install_packages(self._ubuntu_packages)
+            self.__execute_assert_zero("pip3 install --upgrade meson", cwd)
+            self.__execute_assert_zero("mv /usr/bin/meson /usr/bin/meson.bak", cwd)
+            self.__execute_assert_zero("ln -s /usr/local/bin/meson /usr/bin/meson", cwd)
+            self.__execute_assert_zero("pip3 install --upgrade ninja", cwd)
+            self.__execute_assert_zero(
+                "modprobe -a rdma_cm rdma_ucm ib_core ib_uverbs mlx4_core mlx4_ib", cwd
+            )
+
+        elif isinstance(node.os, Redhat):
+            self.__execute_assert_zero(
+                "yum update -y --disablerepo='*' --enablerepo='*microsoft*'", cwd
+            )
+            node.os.group_install_packages("Infiniband Support")
+            node.os.install_packages(self._redhat_packages)
+            self.__execute_assert_zero(
+                "modprobe -a ib_core ib_uverbs mlx4_en mlx4_core mlx5_core mlx4_ib"
+                + " mlx5_ib",
+                cwd,
+            )  # add mellanox drivers
+            self.__execute_assert_zero("systemctl enable rdma", cwd)
+            self.__execute_assert_zero("pip3 install --upgrade meson", cwd)
+            self.__execute_assert_zero("ln -s /usr/local/bin/meson /usr/bin/meson", cwd)
+
+            wget_tool = self.node.tools[Wget]
+            wget_tool.get(self._ninja_url)
+            node.execute(f"mv ninja-linux.zip {node.working_path}/")
+            self.__execute_assert_zero(
+                "unzip ninja-linux.zip && mv ninja /usr/bin/ninja", cwd
+            )
+            self.__execute_assert_zero("pip3 install --upgrade pyelftools", cwd)
+        else:
+            raise SkippedException(
+                f"This os {node.os} is not implemented by the DPDK suite."
+            )
+        return True
+
+    def _generate_testpmd_command(self, nic_to_include: str) -> str:
+        return (
+            f"{self._testpmd_install_path} -l 2,3 -n 4 --proc-type=primary "
+            + f"{nic_to_include} -- --forward-mode=txonly -a --stats-period 1"
+        )
+
+    def run_with_timeout(self, nic_to_include: str, timeout: int) -> str:
+        timer_proc = self.node.execute_async(
+            f"sleep {timeout} && killall -s INT {self._testpmd_install_path}",
+            sudo=True,
+            shell=True,
+        )
+        testpmd_proc = self.node.execute_async(
+            self._generate_testpmd_command(nic_to_include),
+            sudo=True,
+        )
+        timer_proc.wait_result()
+        proc_result = testpmd_proc.wait_result()
+        return proc_result.stdout
+
+    def get_tx_pps_from_testpmd_output(self, output: str) -> int:
+        matches = re.findall(r"Tx-pps:\s+([0-9]+)", output)
+        assert_that(len(matches)).is_not_zero()
```

Comment:
1. add comments to explain it, once it's failed.
2. try `assert_that(matches).is_length(0)`

Reply:
there isn't a hard length requirement, length just has to be larger than 0. I'll add a better description.